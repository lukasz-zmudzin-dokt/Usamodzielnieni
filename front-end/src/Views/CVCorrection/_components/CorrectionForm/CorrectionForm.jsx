import React, { useState } from "react";
import FormGroup from "components/FormGroup";
import { Form, Button, Row, Alert } from "react-bootstrap";
import { sendFeedback } from "Views/CVCorrection/functions";
import { useHistory } from "react-router-dom";

const CorrectionForm = ({ id, token }) => {
  const [fail, setFail] = useState(false);
  const [msg, setMsg] = useState("");
  const [disabled, setDisabled] = useState(false);
  const [feedback, setFeedback] = useState({
    basicInfo: "",
    schools: "",
    experiences: "",
    skills: "",
    languages: "",
    additionalInfo: "",
  });
  const history = useHistory();

  const checkIfNotEmpty = () => {
    for (let key in feedback) {
      if (feedback[key] !== "") {
        return true;
      }
    }
    return false;
  };

  const submit = async (event) => {
    event.preventDefault();
    event.stopPropagation();

    if (checkIfNotEmpty()) {
      setDisabled(true);
      try {
        await sendFeedback(id, token, feedback);
        setFeedback({
          basicInfo: "",
          schools: "",
          experiences: "",
          skills: "",
          languages: "",
          additionalInfo: "",
        });
        setFail(false);
        setMsg("Pomyślnie przesłano uwagi.");
        history.push("/cvApproval");
      } catch (err) {
        setFail(true);
        setMsg("Błąd serwera.");
      }
    } else {
      setFail(true);
      setMsg("Dodaj minimalnie jedną uwagę.");
    }
    setDisabled(false);
  };

  const message = msg ? (
    fail ? (
      <Alert variant="danger">{msg}</Alert>
    ) : (
      <Alert variant="success">{msg}</Alert>
    )
  ) : null;

  return (
    <Form onSubmit={submit} className="CVCorrection__form">
      <FormGroup
        header="Dane osobowe"
        id="basic_info"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, basicInfo: val })}
        val={feedback.basicInfo}
        length={{ min: 1, max: 1000 }}
      />
      <FormGroup
        header="Edukacja"
        id="schools"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, schools: val })}
        val={feedback.schools}
        length={{ min: 1, max: 1000 }}
      />
      <FormGroup
        header="Doświadczenie zawodowe"
        id="experiences"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, experiences: val })}
        val={feedback.experiences}
        length={{ min: 1, max: 1000 }}
      />
      <FormGroup
        header="Umiejętności"
        id="skills"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, skills: val })}
        val={feedback.skills}
        length={{ min: 1, max: 1000 }}
      />
      <FormGroup
        header="Języki obce"
        id="languages"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, languages: val })}
        val={feedback.languages}
        length={{ min: 1, max: 1000 }}
      />
      <FormGroup
        header="Dodatkowe uwagi"
        id="additional_info"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, additionalInfo: val })}
        val={feedback.additionalInfo}
        length={{ min: 1, max: 1000 }}
      />
      {message}
      <Row className="justify-content-center m-0">
        <Button variant="primary" type="submit" disabled={disabled}>
          {disabled ? "Ładowanie..." : "Wyślij uwagi"}
        </Button>
      </Row>
    </Form>
  );
};

export default CorrectionForm;
