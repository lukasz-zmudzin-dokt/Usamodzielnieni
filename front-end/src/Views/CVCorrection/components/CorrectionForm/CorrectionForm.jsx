import React, { useState } from "react";
import FormGroup from "components/FormGroup";
import { Form, Button, Row, Alert } from "react-bootstrap";
import { sendFeedback } from "Views/CVCorrection/functions";

const CorrectionForm = ({ data }) => {
  const [fail, setFail] = useState(false);
  const [msg, setMsg] = useState("");
  const [disabled, setDisabled] = useState(false);
  const [feedback, setFeedback] = useState({
    basicInfo: "",
    schools: "",
    experiences: "",
    skills: "",
    languages: "",
    additionalInfo: "",
  });

  const checkIfNotEmpty = () => {
    for (let key in feedback) {
      if (feedback[key] !== "") {
        return true;
      }
    }
    return false;
  };

  const submit = async (event) => {
    event.preventDefault();
    event.stopPropagation();

    if (checkIfNotEmpty()) {
      setDisabled(true);
      let res;
      try {
        res = sendFeedback(data.id, data.token, feedback);
        setFeedback({
          basicInfo: "",
          schools: "",
          experiences: "",
          skills: "",
          languages: "",
          additionalInfo: "",
        });
        setFail(false);
        setMsg("Pomyślnie przesłano uwagi.");
      } catch (err) {
        setFail(true);
        setMsg("Błąd serwera.");
      }
    } else {
      setFail(true);
      setMsg("Dodaj minimalnie jedną uwagę.");
    }
    setDisabled(false);
  };

  const message = msg ? (
    fail ? (
      <Alert variant="danger">{msg}</Alert>
    ) : (
      <Alert variant="success">{msg}</Alert>
    )
  ) : null;

  return (
    <Form onSubmit={submit} className="CVCorrection__form">
      <FormGroup
        header="Dane osobowe"
        id="basic_info"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, basicInfo: val })}
        val={feedback.basicInfo}
      />
      <FormGroup
        header="Edukacja"
        id="schools"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, schools: val })}
        val={feedback.schools}
      />
      <FormGroup
        header="Doświadczenie zawodowe"
        id="experiences"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, experiences: val })}
        val={feedback.experiences}
      />
      <FormGroup
        header="Umiejętności"
        id="skills"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, skills: val })}
        val={feedback.skills}
      />
      <FormGroup
        header="Języki obce"
        id="languages"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, languages: val })}
        val={feedback.languages}
      />
      <FormGroup
        header="Dodatkowe uwagi"
        id="additional_info"
        type="textarea"
        setVal={(val) => setFeedback({ ...feedback, additionalInfo: val })}
        val={feedback.additionalInfo}
      />
      {message}
      <Row className="justify-content-end m-0">
        <Button
          variant="primary"
          type="submit"
          className=""
          disabled={disabled}
        >
          {disabled ? "Ładowanie..." : "Wyślij uwagi"}
        </Button>
      </Row>
    </Form>
  );
};

export default CorrectionForm;
