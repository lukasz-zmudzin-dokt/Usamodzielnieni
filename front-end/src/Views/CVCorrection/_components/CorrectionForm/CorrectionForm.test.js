import React from "react";
import {
  render,
  fireEvent,
  waitForElement,
  wait,
} from "@testing-library/react";
import CorrectionForm from "./CorrectionForm";

describe("CorrectionForm", () => {
  let data = { id: "abc", token: "abc" };
  it("should match snapshot", () => {
    const { container } = render(<CorrectionForm data={data} />);
    expect(container).toMatchSnapshot();
  });
});
