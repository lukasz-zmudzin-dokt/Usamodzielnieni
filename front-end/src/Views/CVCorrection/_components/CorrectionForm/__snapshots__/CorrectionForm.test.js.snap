// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CorrectionForm should match snapshot 1`] = `
<div>
  <form
    class="CVCorrection__form"
  >
    <div
      class="offerForm__textContainer form-group"
    >
      <label
        class="form-label"
        for="basic_info"
      >
        Dane osobowe
      </label>
      <textarea
        class="offerForm__textarea form-control"
        id="basic_info"
        maxlength="50"
        minlength="1"
      />
    </div>
    <div
      class="offerForm__textContainer form-group"
    >
      <label
        class="form-label"
        for="schools"
      >
        Edukacja
      </label>
      <textarea
        class="offerForm__textarea form-control"
        id="schools"
        maxlength="50"
        minlength="1"
      />
    </div>
    <div
      class="offerForm__textContainer form-group"
    >
      <label
        class="form-label"
        for="experiences"
      >
        Doświadczenie zawodowe
      </label>
      <textarea
        class="offerForm__textarea form-control"
        id="experiences"
        maxlength="50"
        minlength="1"
      />
    </div>
    <div
      class="offerForm__textContainer form-group"
    >
      <label
        class="form-label"
        for="skills"
      >
        Umiejętności
      </label>
      <textarea
        class="offerForm__textarea form-control"
        id="skills"
        maxlength="50"
        minlength="1"
      />
    </div>
    <div
      class="offerForm__textContainer form-group"
    >
      <label
        class="form-label"
        for="languages"
      >
        Języki obce
      </label>
      <textarea
        class="offerForm__textarea form-control"
        id="languages"
        maxlength="50"
        minlength="1"
      />
    </div>
    <div
      class="offerForm__textContainer form-group"
    >
      <label
        class="form-label"
        for="additional_info"
      >
        Dodatkowe uwagi
      </label>
      <textarea
        class="offerForm__textarea form-control"
        id="additional_info"
        maxlength="50"
        minlength="1"
      />
    </div>
    <div
      class="justify-content-end m-0 row"
    >
      <button
        class="btn btn-primary"
        type="submit"
      >
        Wyślij uwagi
      </button>
    </div>
  </form>
</div>
`;
